### ISSUE TITLE
Improve `Support for nested build without a settings` deprecation warning 

### ISSUE THREAD

**Issue by lacasseio** [gradle/gradle#5371]
### Expected Behavior

`gradle init` in an empty folder shouldn't generate deprecation warning.

### Current Behavior

```
$ gradle init --warning-mode=all
Support for nested build without a settings file was deprecated and will be removed in Gradle 5.0. You should create a empty settings file in /path/to/project
```

### Context
Gradle 4.8

**Comment by marcphilipp** [comment]
Looks like this deprecation message was introduced as part of #3137 for 4.4.

**Comment by blindpirate** [comment]
Thanks @lacasseio, I guess some ancestor directories of your init directory contain a `settings.gradle` file.

For example, run `gradle init` in childDir of

```
+-- parentDir
  +-- childDir
```
won't cause a deprecation warning, but run `gradle init` in childDir of

```
+-- parentDir
  +-- settings.gradle
  +-- childDir
```

will cause such warning. That's expected behavior. However, if this even makes you confused, I guess we should improve the warning message to at least make users aware of what happens here. I'll keep it open.

**Comment by lacasseio** [comment]
Good point, I have a settings file in a parent directory. Given using this `settings.gradle` file is not expected in this use case, we could live with the deprecation warning wait until the next major release to have this fixed automatically.

**Comment by oehme** [comment]
It won't be fixed automatically, it'll fail in 5.0. So we definitely need to change this so that `gradle init` doesn't trigger this warning.
